runtime/resolvers/metrics.go

```diff
@@ -142,9 +146,37 @@ func (r *metricsResolver) ResolveInteractive(ctx context.Context) (runtime.Resol
 	if err != nil {
 		return nil, err
 	}
-	return runtime.NewDriverResolverResult(res), nil
+	return runtime.NewDriverResolverResult(res, r.meta), nil
 }
 
 func (r *metricsResolver) ResolveExport(ctx context.Context, w io.Writer, opts *runtime.ResolverExportOptions) error {
 	return errors.New("not implemented")
 }
+
+// metaFromMetricsView extracts metadata from the metrics view.
+func metaFromMetricsView(spec *runtimev1.MetricsViewSpec) []map[string]any {
+	if spec == nil {
+		return nil
+	}
+
+	details := make([]map[string]any, 0)
+	for _, m := range spec.Measures {
```

Comment:
This function should take the `r.query` into account. I.e. it should only include those dimensions and measures that are present in `r.query.Dimensions` and `r.query.Measures`.